sdk/core/Azure.Core.Experimental/src/Serialization/JsonObjectSerializer.cs

```diff
@@ -59,5 +67,85 @@ public override async ValueTask<object> DeserializeAsync(Stream stream, Type ret
         {
             return await JsonSerializer.DeserializeAsync(stream, returnType, _options, cancellationToken).ConfigureAwait(false);
         }
+
+        /// <inheritdoc/>
+        public string? GetSerializedMemberName(MemberInfo memberInfo)
+        {
+            Argument.AssertNotNull(memberInfo, nameof(memberInfo));
+
+            if (!_cache.TryGetValue(memberInfo.ReflectedType, out MemberInfo[] members))
+            {
+                members = GetMembers(memberInfo.ReflectedType).ToArray();
+            }
+
+            foreach (MemberInfo member in members)
+            {
+                if (member == memberInfo)
+                {
+                    return GetPropertyName(memberInfo);
+                }
+            }
+
+            return null;
+        }
+
+        private static IEnumerable<MemberInfo> GetMembers(Type type)
```

Comment:
Why do we need to enumerate instead of just looking at attributes of the provided member?

Reply:
For now that may work, but it won't for STJ in .NET 5 where both properties and fields are serializable (same with Json.NET) and there's logic to determine if overrides should be de/serialized or not. So I can change this now, but for .NET 5 support we'll have to change it back. I can't implement it now because it requires referencing an attribute that doesn't yet exist...though, I suppose we can check it by name like we've done elsewhere in code, but .NET 5 hasn't GA'd yet either.

Reply:
Per offline discussion, I will simplify the code for now, open an issue for tracking support for STJ in .NET 5, and add a comment to the code referencing the issue / summary.

Reply:
> For now that may work, but it won't for STJ in .NET 5 where both properties and fields are serializable

Just an FYI, serializing fields is opt-in based on an option or based on the new `JsonIncludeAttribute`, and isn't on by default (for backwards compatibility). Not sure if that changes anything.
https://github.com/dotnet/runtime/issues/34558#issuecomment-613100180

Reply:
I was aware of that, but if someone opts in we'll need to support that. We could do so by checking attributes' assembly/type names without binding to .NET 5.